cpp/src/arrow/compute/kernels/scalar_set_lookup_test.cc

```diff
@@ -185,25 +244,43 @@ TYPED_TEST(TestIsInKernelPrimitive, IsIn) {
             /*skip_nulls=*/false);
   CheckIsIn(type, "[null, 1, 2, 3, 2]", "[2, 1]", "[false, true, true, false, true]",
             /*skip_nulls=*/true);
+  CheckIsIn(type, "[null, 1, 2, 3, 2]", "[2, 1]", "[null, true, true, false, true]",
```

Comment:
For completeness, I think we should:
- repeat each `skip_null=false` check using `null_matching_behavior=SetLookupOptions::NullMatchingBehavior::MATCH`
- repeat each `skip_null=true` check using `null_matching_behavior=SetLookupOptions::NullMatchingBehavior::SKIP`

Reply:
Sure, have added